src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java

```diff
@@ -38,95 +32,58 @@ public Bank mapRow(ResultSet rs, int rowNum) throws SQLException {
+                    "select * from BANKS where id = %d", id), new RowMapper<Bank>() {
+                @Override
+                public Bank mapRow(ResultSet rs, int i) throws SQLException {
+                    String name = rs.getString("name");
+                    String address = rs.getString("address");
+                    return new Bank(id, name, address);
+                }
+            });
         } catch (Exception e) {
-            e.printStackTrace();
-        } finally {
-            dataBaseUtil.closeConections(con);
+            throw new EntityNotFoundException(String.format(
+                    "Bank with ID = %d does not exist", id));
         }
-        return null;
     }
 
     @Override
     public List<Bank> findAllByName(String name) {
-        List<Bank> banks = new ArrayList();
-        Connection con = dataBaseUtil.startConnection();
         try {
-            PreparedStatement pstmt = con.prepareStatement(String.format("select * from BANKS where name = %s", name));
-            ResultSet rs = pstmt.executeQuery();
-            while (rs.next()) {
-                Integer id = rs.getInt("id");
+        return jdbcTemplate.query(String.format(
+                "select * from BANKS where name = %s", name), new RowMapper<Bank>() {
+            @Override
+            public Bank mapRow(ResultSet rs, int rowNum) throws SQLException {
+                int id = rs.getInt("id");
                 String address = rs.getString("address");
-                Bank bank = new Bank(id, name, address);
-                banks.add(bank);
-            }
-            if (banks.isEmpty()) {
-                throw new EntityNotFoundException(String.format("Bank with name = %s does not exist", name));
+                return new Bank(id, name, address);
             }
-            return banks;
-        } catch (Exception e) {
-            e.printStackTrace();
-        } finally {
-            dataBaseUtil.closeConections(con);
-        }
-        return banks;
+        });
+        }catch (Exception e){
+            throw new EntityNotFoundException(String.format("Bank with name = %s does not exist", name));
+         }
     }
 
 
     @Override
     public void store(Bank bank) {
-        Connection con = dataBaseUtil.startConnection();
-        try {
-            String name = bank.getName();
-            String address = bank.getAddress();
-
-            String query = String.format("INSERT INTO BANKS (name, address) VALUES (\"%s\",\"%s\")", name, address);
-            PreparedStatement pstmt = con.prepareStatement(query);
-            pstmt.executeUpdate();
-
-        } catch (SQLException e) {
-            e.printStackTrace();
-        } finally {
-            dataBaseUtil.closeConections(con);
-        }
+        String name = bank.getName();
+        String address = bank.getAddress();
+        jdbcTemplate.update(String.format(
+                "INSERT INTO BANKS (name, address) VALUES (\"%s\",\"%s\")", name, address));
```

Comment:
`jdbcTemplate.update()` method accepts the query with parameters so you can avoid the `String.format()`.

For instance, that code could be replaced by:

```
jdbcTemplate.update("INSERT INTO BANKS (name, address) VALUES (?, ?)", name, address);
```

Reply:
Done, but seems to work only for jdbcTemplate.update()